docs/pages/architecture/tls-routing.mdx

```diff
@@ -25,6 +25,33 @@ To implement TLS routing, Teleport uses SNI ([Server Name Indication](https://en
 and ALPN ([Application-Level Protocol Negotiation](https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation))
 TLS extensions.
 
+<Admonition
+  type="warning"
+  title="TLS routing is incompatible with load balancers and reverse proxies"
+>
+Using Teleport's TLS routing mode behind a layer 7 (HTTP/HTTPS) proxy is generally not supported, due to
+these proxies terminating TLS themselves and then rewriting their requests to the upstream service, stripping
+the additional SNI/ALPN parts of the request in the process. In order for ALPN to work correctly, the Teleport
+proxy must terminate TLS itself.
+
+Broadly, this means that Teleport's TLS routing functionality is incompatible with:
+- AWS ALBs (Application Load Balancers)
+- AWS NLBs (Network Load Balancers), when using a TLS listener and a public ACM (Amazon Certificate Manager) certificate
+- Commonly used HTTP reverse proxies including nginx, Apache, Caddy, Traefik, HAProxy and many others
+- Cloudflare tunnels in their default configuration
+
+Deploying Teleport in TLS routing mode behind an HTTP proxy will result in a Teleport web UI experience that seems
+to work perfectly, but the use of `tsh`, `tctl` and attempting to join remote Teleport services to the cluster will fail
+with errors like `ssh: overflow reading version string` and `EOF`. A functioning Teleport web UI is not always an indication
+of a correctly configured Teleport cluster.
+
+If in doubt, remove all load balancers/proxies from the equation and connect to Teleport's web port directly to isolate the issue.
+
+To use Teleport behind a reverse proxy, you should either:
+- use a layer 4 (TCP) proxy which forwards TCP streams directly to Teleport (which will in turn handle TLS termination itself)
+- deploy Teleport in its [separate ports](../reference/networking.mdx#ports-without-tls-routing) mode by adding `version: v1` to `/etc/teleport.yaml` and removing `proxy_listener_mode: multiplex`
```

Comment:
It's a bit of a random musing - but - if this is a supported configuration which we are recommending for environments with a L7 LB, we really need to introduce a way to put Teleport into "ports separate" mode that isn't "set your config to V1" as this is going to become more and more problematic as we start introducing future configuration versions. What if I want port separate mode and separate configuration of proxy and auth server addresses - I'll see if we have an outstanding issue for this and raise it.

Reply:
Being totally honest, I hated that we made TLS routing the default mode for exactly this reason - at least 50% of our AWS customers are using ACM and many other customers use reverse proxies. I would personally say we always need to maintain, support and test a non-TLS routing way of deploying Teleport. I would love to see better support for this.